backend/persistent/datastores/drivers/AWSAthenaDriver.js

```diff
@@ -0,0 +1,222 @@
+'use strict';
+
+const AWS = require('aws-sdk');
+import Logger from '../../../logger';
+const NUMBER_OF_RETRIES = 5;
+
+/**
+ * The following function will create an AWS Athena Client
+ * @param {object} connection - AWS Athena Connection Parameters
+ * @param {string} connection.accessKey - AWS Access Key
+ * @param {string} connection.secretAccessKey - AWS Secret Key
+ * @param {string} connection.region - AWS Region
+ * @returns {object} AWS Athena Client
+ */
+export function createAthenaClient(connection) {
+    const connectionParams = {
+        apiVersion: '2017-05-18',
+        accessKeyId: connection.accessKey,
+        secretAccessKey: connection.secretAccessKey,
+        region: connection.region,
+        maxRetries: 5
+    };
+    const athenaClient = new AWS.Athena(connectionParams);
+
+    return athenaClient;
+}
+
+/**
+ * The following method will execute the sql statement to query the
+ * athena database
+ * @param {object} athenaClient  - Object created using create athena client
+ * @param {object} params - Connection Parameters
+ * @param {string} params.dbName - Database name to connect to
+ * @param {string} params.sqlStatement - SQL statement to execute
+ * @param {string} params.s3Outputlocation - Location will Athena will output resutls of query
+ * @return {string} requestId
+ */
+export function startQuery(athenaClient, params) {
+    const client = athenaClient;
+
+    const queryParams = {
+        QueryString: params.sqlStatement,
+        ResultConfiguration: {
+          OutputLocation: params.s3Outputlocation,
+          EncryptionConfiguration: {
+            EncryptionOption: 'SSE_S3'
+          }
+        },
+        QueryExecutionContext: {
+          Database: params.database
+        }
+    };
+    return new Promise(function(resolve, reject) {
+        return client.startQueryExecution(queryParams, (err, data) => {
+            if (err) {
+                Logger.log(`Unexpected Error starting Athena Query ${err}`);
+                return reject(err);
+            }
+                const queryId = data.QueryExecutionId;
+                return resolve(queryId);
+          });
+    });
+}
+
+/**
+ * The following method will check to see if the query results
+ * have completed.  It will return -1 if the query errored, 0
+ * if it is still executing or 1 if it has completed
+ * @param {object} athenaClient  - Object created using create athena client
+ * @param {string} queryExecutionId - AWS Query Execution Id
+ * @returns {int} -1 : Error, 0 : Still running, 1 : Completed
+ */
+export function queryResultsCompleted(athenaClient, queryExecutionId) {
```

Comment:
I'd get rid of this function, because the only thing it does is translate status strings (e.g. `SUCCEEDED`) into numbers (i.e. -1, 0 or 1).

From a logging point of view is better that we call `client.getQueryExecution` directly and:
- `reject(err)` if err
- invoke `getQueryResults` if `status === SUCCEEDED`
- retry `client.getQueryExecution` if `status === QUEUED or RUNNING`
- in any other case with `reject(new Error('Query failed with status ${status}');`